sdk/eventgrid/azure-eventgrid/azure/eventgrid/_publisher_client.py

```diff
@@ -113,27 +117,24 @@ def send_events(self, events, **kwargs):
         if not isinstance(events, list):
             events = cast(ListEventType, [events])
 
-        if all(isinstance(e, CloudEvent) for e in events) or all(_is_cloud_event(e) for e in events):
+        if isinstance(events[0], CloudEvent) or _is_cloud_event(events[0]):
             try:
                 events = [cast(CloudEvent, e)._to_generated(**kwargs) for e in events] # pylint: disable=protected-access
             except AttributeError:
                 pass # means it's a dictionary
             kwargs.setdefault("content_type", "application/cloudevents-batch+json; charset=utf-8")
-            self._client.publish_cloud_event_events(
+            return self._client.publish_cloud_event_events(
                 self._endpoint,
                 cast(List[InternalCloudEvent], events),
                 **kwargs
                 )
-        elif all(isinstance(e, EventGridEvent) for e in events) or all(isinstance(e, dict) for e in events):
-            kwargs.setdefault("content_type", "application/json; charset=utf-8")
+        kwargs.setdefault("content_type", "application/json; charset=utf-8")
+        if isinstance(events[0], EventGridEvent) or _is_eventgrid_event(events[0]):
             for event in events:
                 _eventgrid_data_typecheck(event)
-            self._client.publish_events(self._endpoint, cast(List[InternalEventGridEvent], events), **kwargs)
```

Comment:
did you remove this line with `publish_events` for EventGridEvents since `publish_custom_event_events` accomplishes the same thing?

Reply:
that's right - this was we are able to allow a dict representation for a customevent  as well

Reply:
I'm assuming that this means `publish_custom_event_events` and `publish_events` are interchangeable. And, in that case, is there a need for separate `publish_events` and `publish_custom_event_events` methods? 

would it be better to only have the `publish_events` method generated by the swagger since it can be used for sending both eg events and custom events? Or is that not something we want to do because we shouldn't change the swagger since it is used by all languages?

Reply:
^^ to address above and summarize discussion with @rakshith91 : these are not the same methods, since they accept different types of objects. We can use `publish_custom_event_events` for sending both since it accepts `List[object]`. Can't use `publish_events` for both since it only accepts `List[EventGridEvent]`. The separate methods are generated by the swagger and needed for other languages.